app/src/main/java/com/example/habitsmasher/LoginFragment.java

```diff
@@ -217,13 +217,16 @@ public void onSuccess(DocumentSnapshot documentSnapshot) {
      * @param googleUser the user to add
      */
     private void addUserToDatabaseIfUserDoesNotExist(User googleUser) {
+        Log.d(TAG, "addUserToDatabaseIfUserDoesNotExist: entry");
         FirebaseFirestore.getInstance()
                          .collection("Users")
                          .document(googleUser.getId()).get().addOnCompleteListener(
                 new OnCompleteListener<DocumentSnapshot>() {
                     @Override
                     public void onComplete(@NonNull Task<DocumentSnapshot> task) {
-                        if (!task.isSuccessful()) {
+                        Log.d(TAG, "addUserToDatabaseIfUserDoesNotExist: checking if user exists");
+                        if (!task.getResult().exists()) {
```

Comment:
This was the line that was wrong...

Before: `if !task.isSuccessful()` was incorrect because it is always successful... but we should check if it **exists** instead

Now: `if (!task.getResult().exists()`